lib/web_ui/lib/src/ui/hash_codes.dart

```diff
@@ -4,96 +4,91 @@
 
 part of ui;
 
-class _HashEnd {
-  const _HashEnd();
-}
-
+class _HashEnd { const _HashEnd(); }
 const _HashEnd _hashEnd = _HashEnd();
 
-// Jenkins hash function, optimized for small integers.
+/// Jenkins hash function, optimized for small integers.
 //
 // Borrowed from the dart sdk: sdk/lib/math/jenkins_smi_hash.dart.
-abstract class _Jenkins {
-  // This class is not meant to be instantiated or extended; this constructor
-  // prevents instantiation and extension.
-  _Jenkins._();
-
+class _Jenkins {
   static int combine(int hash, Object? o) {
     assert(o is! Iterable);
     hash = 0x1fffffff & (hash + o.hashCode);
     hash = 0x1fffffff & (hash + ((0x0007ffff & hash) << 10));
-    hash = hash ^ (hash >> 6);
-    return hash;
+    return hash ^ (hash >> 6);
   }
 
   static int finish(int hash) {
     hash = 0x1fffffff & (hash + ((0x03ffffff & hash) << 3));
     hash = hash ^ (hash >> 11);
-    hash = 0x1fffffff & (hash + ((0x00003fff & hash) << 15));
-    return hash;
+    return 0x1fffffff & (hash + ((0x00003fff & hash) << 15));
   }
 }
 
+/// Combine up to twenty objects' hash codes into one value.
+///
+/// If you only need to handle one object's hash code, then just refer to its
+/// [Object.hashCode] getter directly.
+///
+/// If you need to combine an arbitrary number of objects from a [List] or other
+/// [Iterable], use [hashList]. The output of [hashList] can be used as one of
+/// the arguments to this function.
+///
+/// For example:
+///
+/// ```dart
+/// int hashCode => hashValues(foo, bar, hashList(quux), baz);
+/// ```
+@Deprecated(
+  'Use Object.hash() instead. '
+  'This feature was deprecated in v2.9.0-0.1.pre'
+)
 int hashValues(
```

Comment:
Drive-by comment:
You could possibly just redirect to Object.hash here instead of keeping the implementation.

Same below

Reply:
I tried passing in all the args from `hashValues` to `Object.hash`, but as pointed out in the above thread (unresolved it now), it will provide inconsistent hash values. If you are implying a different way to redirect, could you elaborate?

Reply:
Ah sorry - I had not seen that discussion - no I did not have anything more clever in mind :)